library/cartridge_check_instance_started.py

```diff
@@ -4,32 +4,94 @@
 from ansible.module_utils.helpers import ModuleRes, CartridgeException
 from ansible.module_utils.helpers import get_control_console
 
-
 argument_spec = {
     'control_sock': {'required': True, 'type': 'str'},
     'stateboard': {'required': True, 'type': 'bool'},
+    'expected_states': {'required': False, 'type': 'list', 'default': ['Unconfigured', 'RolesConfigured']},
+    'check_buckets_are_discovered': {'required': False, 'type': 'bool', 'default': False},
 }
 
 
-def check_instance_started(params):
-    try:
-        control_console = get_control_console(params['control_sock'])
-        instance_is_alive = True
-        if not params['stateboard']:
-            instance_is_alive, _ = control_console.eval_res_err('''
-                return require('membership').myself().status == 'alive'
+def check_stateboard_started(control_console):
+    box_status, err = control_console.eval_res_err('''
+        if type(box.cfg) == 'function' then
+            return nil, "Box hasn't been configured"
+        end
+        return box.info.status
+    ''')
```

Comment:
I'd suggest to check `box.cfg.listen ~= nil`. See the `stateboard.cfg()` implementation: https://github.com/tarantool/cartridge/blob/master/cartridge/stateboard.lua#L386-L387. It also overlaps `box.info.status` check